app/controllers/catalog_controller.rb

```diff
@@ -2199,6 +2213,38 @@ def fetch_playbook_details
     playbook_details
   end
 
+  def fetch_terraform_template_details
+    terraform_template_details = {}
+    provision = @record.config_info[:provision]
+    terraform_template_details[:provisioning] = {}
+    terraform_template_details[:provisioning][:repository] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptSource, provision[:repository_id])
+    terraform_template_details[:provisioning][:template] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptPayload, provision[:configuration_script_payload_id])
+    terraform_template_details[:provisioning][:credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential, provision[:credential_id])
+    terraform_template_details[:provisioning][:network_credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::NetworkCredential, provision[:network_credential_id]) if provision[:network_credential_id]
+    fetch_dialog(terraform_template_details, provision[:dialog_id], :provisioning)
+    terraform_template_details[:provisioning][:execution_ttl] = provision[:execution_ttl]
+    terraform_template_details[:provisioning][:verbosity] = provision[:verbosity]
+    terraform_template_details[:provisioning][:log_output] = provision[:log_output]
+    terraform_template_details[:provisioning][:become_enabled] = provision[:become_enabled] == true ? _('Yes') : _('No')
```

Comment:
Perhaps building up the hash first would make this easier to follow:

```ruby
    provisioning_details = {}    
    provisioning_details[:repository] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptSource, provision[:repository_id])
    provisioning_details[:template] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptPayload, provision[:configuration_script_payload_id])
    provisioning_details[:credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::Credential, provision[:credential_id])
    provisioning_details[:network_credential] = fetch_name_from_object(ManageIQ::Providers::EmbeddedTerraform::AutomationManager::NetworkCredential, provision[:network_credential_id]) if provision[:network_credential_id]
    fetch_dialog(provisioning_details, provision[:dialog_id], :provisioning)
    provisioning_details[:execution_ttl] = provision[:execution_ttl]
    provisioning_details[:verbosity] = provision[:verbosity]
    provisioning_details[:log_output] = provision[:log_output]
    provisioning_details[:become_enabled] = provision[:become_enabled] == true ? _('Yes') : _('No')

    terraform_template_details[:provisioning] = provisioning_details
```

Reply:
what's `fetch_dialog(provisioning_details, provision[:dialog_id], :provisioning` doing in the middle of this block of hash assignment?  Can it occur later or before?

Reply:
This was just copy pasted from the ansible function, I can move it to the end for both functions.